src/engine/GroupBy.h

```diff
@@ -157,49 +157,68 @@ class GroupBy : public Operation {
   // `?z`.
   bool computeGroupByForJoinWithFullScan(IdTable* result);
 
-  // TODO: Update documentation
+  // Create a HashMap containing the results of each aggregation for each group.
```

Comment:
That's not quite true, you only internally create the hash map, and then use it to compute the result. 
This function returns the IDTable, not the hash map:)